meta-lmp-base/classes/fip-utils.bbclass

```diff
@@ -134,6 +134,13 @@ do_deploy:append:class-target() {
             FIP_FWCONFIG="--fw-config ${FIP_DEPLOYDIR_FWCONF}/${dt}-${FIP_FW_CONFIG}-${config}.${FIP_FW_CONFIG_SUFFIX}"
             # Init FIP hw-config settings
             [ -f "${FIP_DEPLOYDIR_UBOOT}/${FIP_UBOOT_DTB}.${FIP_UBOOT_DTB_SUFFIX}" ] || bbfatal "Missing ${FIP_UBOOT_DTB}.${FIP_UBOOT_DTB_SUFFIX} file in folder: ${FIP_DEPLOYDIR_UBOOT}"
+            # Add boot firmware version to U-Boot DTB (if it's defined and is not zero)
+            if [ -n "${LMP_BOOT_FIRMWARE_VERSION}" -a "${LMP_BOOT_FIRMWARE_VERSION}" != "0" ]; then
+                # Might return "FDT_ERR_EXISTS" error, if "lmp" node already exists
+                fdtput -c -t s "${FIP_DEPLOYDIR_UBOOT}/${FIP_UBOOT_DTB}.${FIP_UBOOT_DTB_SUFFIX}" /firmware/bootloader && true
+                fdtput -t s "${FIP_DEPLOYDIR_UBOOT}/${FIP_UBOOT_DTB}.${FIP_UBOOT_DTB_SUFFIX}" /firmware/bootloader compatible "lmp,bootloader"
```

Comment:
One question I had is that this would end up causing an issue since this class is changing the content of a file that gets deployed by the u-boot recipe.

Wonder if we should instead just copy the original file and then modify as needed (and use the new version when creating the fit). @quaresmajose what do you think?

Reply:
addressed

Reply:
Yeah, caching is indeed interesting, and I guess that is why for signing the kernel fit it basically references a file that gets installed by another recipe (u-boot dtb, provided via STAGING_DATADIR), and then it deploys as part of the kernel recipe, instead of changing a file that was deployed by another file.